### PR TITLE
removal rewriter/AstWalker/AstVisitor

### PR DESCRIPTION
**Pull request by yliuuuu**
## Relevant Issues
- [Related To] Issue #682 

## Description
- Removed:
   - AstRewriter Interface & AstRewriterBase class
   - AstVisitor Interface & AstVisitorBase class
   - AstWalker class
   - MetaStrippingRewriter class
   - AstRewriterBaseTest
   - AstWalkerTest
   - RewriterTestBase
   - Tests in ThreadInterruptedTests that tests on AstVisitor and AstWalker
   - AstRewriterBaseTestAdapter
   - ExprNodeTestCase
   - EvaluatorTestCasesAsExprNodeTestCases
   - PartiqlAstToExprNodeRoundTripTests (This is removed here because it has dependency on EvaluatorTestCasesAsExprNodeTestCases)

Note that to achieve the same functionality in testing, the followings are added:
- EvaluatorTestCasesAsStatementTestCases
- StatementTestCases
- VisitorTransformTestAdapter
- VisitorTransformBaseTest

## Other Information
- Updated Unreleased Section in CHANGELOG: **[YES/NO]**
  - Yes
- Any backward-incompatible changes? **[YES/NO]**
  - Yes. The deprecated rewriter, astWalker, and astVisitor will be removed from our code base. 

- Any new external dependencies? **[YES/NO]**
  - No

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/test/org/partiql/lang/ast/passes/AstRewriterBaseTest.kt [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-class AstRewriterBaseTest {
-    @Suppress("DEPRECATION")
-    private val defaultRewriter = AstRewriterBase()
-    @ParameterizedTest
-    @ArgumentsSource(EvaluatorTestCasesAsExprNodeTestCases::class)
-    fun identityRewriteTest(tc: ExprNodeTestCase) {
-        val rewritten = defaultRewriter.rewriteExprNode(tc.expr)
-        tc.assertEquals(rewritten)
-    }
-}
```
Comment (lziq) [inline lang/test/org/partiql/lang/ast/passes/AstRewriterBaseTest.kt:26]: Do we also want to remove https://github.com/partiql/partiql-lang-kotlin/blob/removal-AstRewriter/lang/test/org/partiql/lang/util/testdsl/ExprNodeTestCase.kt?

File: CHANGELOG.md [hunk 2]
```diff
@@ -41,6 +41,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - MetaContainerImpl
   - MetaDeserializer
   - MemoizedMetaDeserializer
+- Removes the deprecated Rewriter/AstWalker/AstVisitor in the code base, removed interfaces and classes include:
+  - AstRewriter Interface & AstRewriterBase class
+  - AstVisitor Interface & AstVisitorBase class
+  - AstWalker class
+  - MetaStrippingRewriter class
```
Comment (lziq) [inline CHANGELOG.md:48]: I think it's better to use brackets to surround the classes/interfaces referenced from the code, e.g., `AstWalker` -> `[AstWalker]`

**Review by lziq — COMMENTED** [review]
Overall looks good to me. Only a few comments left.

**Comment by lziq** [comment]
Another thing is that instead of deleting `ExprNode` & related API, it's better to first work on removing PartiQL compiler's APIs which are involved with `ExprNode`, e.g., `SqlParser.parseExprNode()`, so only APIs related to conversion between PIG AST & `ExprNode` exist. But I won't block this PR because of this.

**Comment by yliuuuu** [comment]
> Another thing is that instead of deleting `ExprNode` & related API, it's better to first work on removing PartiQL compiler's APIs which are involved with `ExprNode`, e.g., `SqlParser.parseExprNode()`, so only APIs related to conversion between PIG AST & `ExprNode` exist. But I won't block this PR because of this.

I tried this before. Unfortunately, such top down approach, even though it is conceptually clear, leads to a huge diff and making the PR less reviewable IMO.

**Review by lziq — APPROVED** [review]

**Pull request merged**